event-sourcing/event-sourcing-core/src/main/scala/org/apache/james/eventsourcing/EventBus.scala

```diff
@@ -19,22 +19,34 @@
 package org.apache.james.eventsourcing
 
 import javax.inject.Inject
+
 import org.apache.james.eventsourcing.eventstore.{EventStore, EventStoreFailedException}
+import org.reactivestreams.Publisher
 import org.slf4j.LoggerFactory
 
+import reactor.core.scala.publisher.{SFlux, SMono}
+
 object EventBus {
   private val LOGGER = LoggerFactory.getLogger(classOf[EventBus])
 }
 
 class EventBus @Inject() (eventStore: EventStore, subscribers: Set[Subscriber]) {
   @throws[EventStoreFailedException]
-  def publish(events: List[Event]): Unit = {
-    eventStore.appendAll(events)
-    events
-      .flatMap((event: Event) => subscribers.map(subscriber => (event, subscriber)))
-      .foreach {case (event, subscriber) => handle(event, subscriber)}
+  def publish(events: Iterable[Event]): SMono[Void] = {
+    SMono(eventStore.appendAll(events))
+        .`then`(runHandles(events, subscribers))
+
   }
 
+  def runHandles(events: Iterable[Event], subscribers: Set[Subscriber]): SMono[Void] = {
+    SFlux.fromIterable(events.flatMap((event: Event) => subscribers.map(subscriber => (event, subscriber))))
+      .concatMap(infos => runHandle(infos._1, infos._2))
+      .`then`()
```

Comment:
why?

Reply:
To get a `SMono[Void]`

Reply:
Why not?

Reply:
> Why not?

because there's another `then` right after?

Reply:
Because there is no direct way to go from `SFlux[A]` to `SMono[Void]`